tests/connection/connection.cpp

```diff
@@ -0,0 +1,546 @@
+    comp1->addVariable(v1);
+    m.addComponent(comp1);
+    libcellml::Variable::addEquivalence(v1,v2);
+    std::string a = m.serialise(libcellml::FORMAT_XML);
+    EXPECT_EQ(e, a);
+}
+
+TEST(Connection, validConnection) {
+    const std::string e =
+    "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+    "<model xmlns=\"http://www.cellml.org/cellml/1.2#\">"
+      "<component name=\"component1\">"
+        "<variable name=\"variable1\"/>"
+      "</component>"
+      "<component name=\"component2\">"
+        "<variable name=\"variable2\"/>"
+      "</component>"
+      "<connection>"
+        "<map_components component_1=\"component1\" component_2=\"component2\"/>"
+        "<map_variables variable_1=\"variable1\" variable_2=\"variable2\"/>"
+      "</connection>"
+    "</model>";
+    libcellml::Model m;
+    libcellml::ComponentPtr comp1 = std::make_shared<libcellml::Component>();
+    libcellml::ComponentPtr comp2 = std::make_shared<libcellml::Component>();
+    libcellml::VariablePtr v1 = std::make_shared<libcellml::Variable>();
+    libcellml::VariablePtr v2 = std::make_shared<libcellml::Variable>();
+    comp1->setName("component1");
+    comp2->setName("component2");
+    v1->setName("variable1");
+    v2->setName("variable2");
+    comp1->addVariable(v1);
+    comp2->addVariable(v2);
+    m.addComponent(comp1);
+    m.addComponent(comp2);
+    libcellml::Variable::addEquivalence(v1,v2);
+    std::string a = m.serialise(libcellml::FORMAT_XML);
+    EXPECT_EQ(e, a);
+}
+
+TEST(Connection, twoMapVariablesConnection) {
+    const std::string e =
+    "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+    "<model xmlns=\"http://www.cellml.org/cellml/1.2#\">"
+      "<component name=\"component1\">"
+        "<variable name=\"variable11\"/>"
+        "<variable name=\"variable12\"/>"
+      "</component>"
+      "<component name=\"component2\">"
+        "<variable name=\"variable21\"/>"
+        "<variable name=\"variable22\"/>"
+      "</component>"
+      "<connection>"
+        "<map_components component_1=\"component1\" component_2=\"component2\"/>"
+        "<map_variables variable_1=\"variable11\" variable_2=\"variable21\"/>"
+        "<map_variables variable_1=\"variable12\" variable_2=\"variable22\"/>"
+      "</connection>"
+    "</model>";
+
+    libcellml::Model m;
+    libcellml::ComponentPtr comp1 = std::make_shared<libcellml::Component>();
+    comp1->setName("component1");
+    libcellml::ComponentPtr comp2 = std::make_shared<libcellml::Component>();
+    comp2->setName("component2");
+    libcellml::VariablePtr v11 = std::make_shared<libcellml::Variable>();
+    v11->setName("variable11");
+    libcellml::VariablePtr v12 = std::make_shared<libcellml::Variable>();
+    v12->setName("variable12");
+    libcellml::VariablePtr v21 = std::make_shared<libcellml::Variable>();
+    v21->setName("variable21");
+    libcellml::VariablePtr v22 = std::make_shared<libcellml::Variable>();
+    v22->setName("variable22");
```

Comment:
I would personally declare everything and then set the names.

Otherwise, I like it that there are empty lines here and there in this test (in fact, I would have added a few more). I find it easier to read the code... (not only the tests, but the code in general)